### ISSUE TITLE
2.3, 2.4 release?

### ISSUE THREAD

**Issue by seratch** [m3dev/octoparts#82]
@lloydmeta asked me to publish scala-ws-client 2.4.1. But I've not figured out how the Octoparts release has been done. I have a couple of questions.

- 2.3 tag exists but [octoparts libs haven't been published yet](http://search.maven.org/#search%7Cga%7C1%7Coctoparts)
- latest version is still 2.4-SNAPSHOT, I couldn't find 2.4 libs in our in-house repository too

So what should I do? publish 2.3 and 2.4? or skip 2.3?

**Comment by lloydmeta** [comment]
Sorry for the confusion :bow: 

You bring up a good point; I think somewhere along the way our server versioning and client versioning diverged but we still had the same version for the overall project (we should come up with a consistent way to handle this), but that doesn't explain why we have only 2.2 client libs released.

My best guess is that the 2.3 tag is for the server side code; the client-side code likely stayed the same, which is why we didn't publish the client libs (we don't publish the server code to Maven because there is no JAR or WAR).

After "releasing" 2.3 (the server) by tagging the release, we moved on to 2.4-SNAPSHOT. I believe we have a few more things we want to put into the 2.4 server-side code before releasing 2.4, so we thought that a 2.4 release doesn't make sense yet.

In summary, for now, maybe we should:

- Publish a 2.3.1 Scala-ws-client instead; add a new tag.
- Start considering separate versions for server and client; keeping in mind that it should be easy to tell which client version corresponds to which server version...

cc @cb372 @mauhiz

**Comment by mauhiz** [comment]
I think we should have the client minor version match the server minor version. Too bad we failed to do it right until now... I'd say let's release a 2.3.1 as @lloydmeta suggested.

**Comment by seratch** [comment]
I'm working on this now.

**Comment by seratch** [comment]
I just found an issue on Scala 2.10 builds.

```
[warn]
[warn] 	Note: Unresolved dependencies path:
[warn] 		org.scaldi:scaldi-play_2.10:0.4.1 (/Users/k-sera/Documents/github/octoparts/project/Common.scala#L59)
[warn] 		  +- com.m3:octoparts_2.10:2.3.1
[trace] Stack trace suppressed: run last octoparts/*:update for the full output.
[error] (octoparts/*:update) sbt.ResolveException: unresolved dependency: org.scaldi#scaldi-play_2.10;0.4.1: not found
```

**Comment by mauhiz** [comment]
Models/client build on 2.10, but server only builds on 2.11

**Comment by seratch** [comment]
I see. `sbt +publishSigned` fails due to that. 
We should have a script to automate publishing libs. I'll do that later.

**Comment by seratch** [comment]
Wow, I found octoparts libs 2.3 here. 
http://search.maven.org/#search%7Cga%7C1%7Cg%3A%22com.m3%22

![screen shot 2015-01-19 at 11 46 58](https://cloud.githubusercontent.com/assets/19658/5795356/997d0528-9fd1-11e4-88af-c17bbd61419c.png)

The `organization` looks broken since 2.3 release.... 
https://github.com/m3dev/octoparts/commit/ee0ca43f#diff-c275e43e5d4ef31cb36384dbbf818da8R17

**Comment by seratch** [comment]
Ah, no. The artifactIds should be fixed.

**Comment by seratch** [comment]
I just fixed artifactId and publishing libs is done. I'll update website now.

**Comment by lloydmeta** [comment]
@seratch thanks a lot for fixing and releasing.

**Comment by seratch** [comment]
Done. >documentation
